### ISSUE TITLE
WT stops cursor blinking after quitting Emacs nox build

### ISSUE THREAD

**Issue by angelog0** [microsoft/terminal#12372]
### Windows Terminal version

1.11.3471.0

### Windows build number

10.0.19044.1503

### Other Software

emacs nox build from master in WSL2
emacs msys2 package (for terminals)

### Steps to reproduce

- Start WT-WSL2: cursor is blinking
- Start Emacs: cursor is blinking
- Quit Emas: at WSL prompt the cursor is no longer blinking.

The same occurs running, in WT, Emacs from MSYS2 package.

NOTICE that THE SAME STEPS in MinTTY terminal (with or without WSL) produce a BLINKING cursor **after quitting Emacs**!.

I see there are similar issue flagged here but in my case the cursor stops blinking ONLY after QUITTING Emacs, not running a curser app...

**Comment by j4james** [comment]
I'm almost sure this is the same thing as #10754, and Emacs is probably using cnorm/cursor_normal on exit, which disables the cursor blinking mode.

I'm not sure why you're getting different results in Mintty, unless you've configured your `TERM` variable to something weird. If you execute `tput cnorm` in Mintty, does that disable blinking?

**Comment by angelog0** [comment]
@j4james, I did not change TERM in MinTTY. It still is

```
$ echo $TERM
xterm
```

BTW, after I quit Emacs (and cursor stops blinking), restarting Emacs the cursor returns to blinking **in Emacs**; exiting one more time, it returns to stop blinking...

**Comment by j4james** [comment]
I think I know what's happening now. It looks like Mintty ignores the blinking escape sequence (private mode 12) if you've set your default cursor preference to be blinking, so that's why it appears to be working. Emacs is still requesting that the cursor stop blinking, but Mintty is simply ignoring that request.

> BTW, after I quit Emacs (and cursor stops blinking), restarting Emacs the cursor returns to blinking in Emacs; exiting one more time, it returns to stop blinking...

That is exactly what I would expect. They're likely using the `cvvis` terminfo capability on startup (which enables cursor blinking), and the `cnorm` capability on shutdown (which disables blinking).

The bottom line is that this is a duplicate of #10754 and is really an Emacs problem. You can probably work around it by editing your terminfo if you don't ever want applications to stop your cursor blinking.

**Comment by angelog0** [comment]
> If you execute `tput cnorm` in Mintty, does that disable blinking?

No...

**Comment by Eli-Zaretskii** [comment]
Is there a way of detecting whether the terminal cursor blinks?  If so, Emacs could do that at startup.

**Comment by j4james** [comment]
@Eli-Zaretskii There is an escape sequence which several terminals support that can be used to reset the cursor to the user's preference - both the shape and blinking. To see how that works, type `printf "\e[ q"` from a WSL shell.

Emacs is unlikely to use that, though, because I suspect they do everything through ncurses, but if they have the concept of shutdown script, you could maybe configure it to output that sequence yourself. That's what many vim users do.

**Comment by zadjii-msft** [comment]
Thanks for the investigation @j4james!

/dup #10754
